evalml/pipelines/pipeline_base.py

```diff
@@ -150,6 +151,34 @@ def linearized_component_graph(self):
         """A component graph in list form. Note that this is not guaranteed to be in proper component computation order"""
         return ComponentGraph.linearized_component_graph(self.component_graph)
 
+    @staticmethod
+    def _make_component_dict_from_component_list(component_list):
```

Comment:
Moved this into pipeline_base... mostly because trying to import this from pipeline.utils.py, which imports PipelineBase, would otherwise create a circular dependency :/